src/ir_Haier.cpp

```diff
@@ -654,27 +654,101 @@ void IRHaierAC176::setMode(uint8_t mode) {
+    if (old_temp == temp) return;
+    if (old_temp > temp)
+      _.Button = kHaierAcYrw02ButtonTempDown;
+    else
+      _.Button = kHaierAcYrw02ButtonTempUp;
+
+    if (degree >= 77) { temp++; }
+    if (degree >= 79) { temp++; }
+    // See at IRHaierAC176::getTemp() comments for clarification
+    _.ExtraDegreeF = temp % 2;
+    _.Temp = (temp - kHaierAcYrw02MinTempF -_.ExtraDegreeF) >> 1;
+    _.UseFahrenheit = true;
+  } else {
+    uint8_t temp = degree;
+    if (temp < kHaierAcYrw02MinTempC)
+      temp = kHaierAcYrw02MinTempC;
+    else if (temp > kHaierAcYrw02MaxTempC)
+      temp = kHaierAcYrw02MaxTempC;
+
+    uint8_t old_temp = getTemp();
+    if (old_temp == temp) return;
+    if (old_temp > temp)
+      _.Button = kHaierAcYrw02ButtonTempDown;
+    else
+      _.Button = kHaierAcYrw02ButtonTempUp;
+    _.Temp = temp - kHaierAcYrw02MinTempC;
+    _.UseFahrenheit = false;
+  }
 }
 
 /// Get the current temperature setting.
 /// @return The current setting for temp. in degrees celsius.
-uint8_t IRHaierAC176::getTemp(void) const { return _.Temp + kHaierAcMinTemp; }
+uint8_t IRHaierAC176::getTemp(void) const {
+  if (!_.UseFahrenheit) { return _.Temp + kHaierAcYrw02MinTempC; }
+  uint8_t degree = _.Temp*2 + kHaierAcYrw02MinTempF + _.ExtraDegreeF;
+  // The way of coding the temperature in degree Fahrenheit is
+  // kHaierAcYrw02MinTempF + Temp*2 + ExtraDegreeF, for example
+  // Temp = 0b0011, ExtraDegreeF = 0b1, temperature is 60 + 3*2 + 1 = 67F
+  // But around 78F there is unconsistency, see table below
+  //
+  // | Fahrenheit | Temp   | ExtraDegreeF |
+  // |    60F     | 0b0000 |     0b0      |
+  // |    61F     | 0b0000 |     0b1      |
+  // |    62F     | 0b0001 |     0b0      |
+  // |    63F     | 0b0001 |     0b1      |
+  // |    64F     | 0b0010 |     0b0      |
+  // |    65F     | 0b0010 |     0b1      |
+  // |    66F     | 0b0011 |     0b0      |
+  // |    67F     | 0b0011 |     0b1      |
+  // |    68F     | 0b0100 |     0b0      |
+  // |    69F     | 0b0100 |     0b1      |
+  // |    70F     | 0b0101 |     0b0      |
+  // |    71F     | 0b0101 |     0b1      |
+  // |    72F     | 0b0110 |     0b0      |
+  // |    73F     | 0b0110 |     0b1      |
+  // |    74F     | 0b0111 |     0b0      |
+  // |    75F     | 0b0111 |     0b1      |
+  // |    76F     | 0b1000 |     0b0      |
+  // |  Not Used  | 0b1000 |     0b1      |
+  // |    77F     | 0b1001 |     0b0      |
+  // |  Not Used  | 0b1001 |     0b1      |
+  // |    78F     | 0b1010 |     0b0      |
+  // |    79F     | 0b1010 |     0b1      |
+  // |    80F     | 0b1011 |     0b0      |
+  // |    81F     | 0b1011 |     0b1      |
+  // |    82F     | 0b1100 |     0b0      |
+  // |    83F     | 0b1100 |     0b1      |
+  // |    84F     | 0b1101 |     0b0      |
+  // |    86F     | 0b1110 |     0b0      |
+  // |    85F     | 0b1101 |     0b1      |
+  if (degree >= 78) { degree--; }
+  if (degree >= 79) { degree--; }
```

Comment:
You've got `77` & `79` in `setTemp()`. Something is probably wrong.

Reply:
The same as above

Reply:
My point was, you've got 77 & 79 in one place, and 78, & 79 in another. If they are both correct, there should be a comment or something to explain that, otherwise someone might make the same conclusion I did.

Reply:
I see your point. If change 78 to 77 in `IRHaier::getTemp()`, code still work correct, tests pass ok. In this case, it is needed some additional explainations in addition to comments in function `IRHaier::getTemp()`?

Reply:
You're the one knowledgeable with the code. I'll let you choose the best code or values for it. I'm just recommending that if the values are going to be different in the two locations, putting a comment in the code is required so someone looking at the code later understands why they are different so they don't try to correct a perceived typo.
e.g. Me, the code reviewer, thought because the values were not the same, that there probably was a mistake.

Reply:
This code will work properly with values 77&79 in both places and using same values in both places will reduce confusion, I'll push this patch later. I asked for you point, because I want to be sure, that it is clear, how this code works. There are enough proposed comments in `setTemp()` for understand, how it works or maybe needed additional explanations in comments?

Reply:
I understood how it works when i first looked at it with the existing comments. It was just the numbers being different that seemed odd.

Reply:
Okey, good! I'll push it later for your review.